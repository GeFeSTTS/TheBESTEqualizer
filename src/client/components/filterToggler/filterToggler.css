.modesElement {
  text-align: center;
  color: var(--mainText-ordinary-white);
}

.labels {
  margin: 0 5px;
}

.switch {
  position: relative;
  display: inline-block;
  width: 46px;
  height: 20px;
  align-self: center;
}

.switch input {
  display: none;
}

.switch input:checked+.slider {
  background-color: var(--togglerSliderBackOn-whitesmoke);
}

.switch input:checked+.slider:before {
  -webkit-transform: translateX(26px);
  -moz-transform: translateX(26px);
  -ms-transform: translateX(26px);
  -o-transform: translateX(26px);
  transform: translateX(26px);
  background: var(--mainText-accent-blue);
}

.slider {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: table-cell;
  height: 20px;
  margin-top: 3px;
  cursor: pointer;
  -webkit-transition: .4s;
  -moz-transition: .4s;
  -ms-transition: .4s;
  -o-transition: .4s;
  transition: .4s;
  vertical-align: middle;
  -webkit-border-radius: 34px;
  -moz-border-radius: 34px;
  border-radius: 34px;
  background-color: var(--rangeSlider-ocean);
  -ms-border-radius: 34px;
  -o-border-radius: 34px;
}

.slider:before {
  position: absolute;
  bottom: 3px;
  left: 4px;
  width: 13px;
  height: 13px;
  content: '';
  -webkit-transition: .4s;
  -moz-transition: .4s;
  -ms-transition: .4s;
  -o-transition: .4s;
  transition: .4s;
  -webkit-border-radius: 50%;
  -moz-border-radius: 50%;
  border-radius: 50%;
  background-color: var(--togglerSliderOff-white);
  -ms-border-radius: 50%;
  -o-border-radius: 50%;
}
