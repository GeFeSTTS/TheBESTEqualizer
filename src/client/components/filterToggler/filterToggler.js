import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './filterToggler.css';
import { attachFiltersToSource, removeSourceFilters } from '../../helpers/equalizerAuxMethods';

export const FilterToggler = (props) => {
  const [toggler, setToggler] = useState(false);
  const {
    audioData: {
      voice,
      sound,
    },
    blocksData,
  } = props;
  const onToggler = () => {
    setToggler(!toggler);
    if (toggler) {
      removeSourceFilters(voice, blocksData);
      attachFiltersToSource(sound, blocksData);
    } else {
      removeSourceFilters(sound, blocksData);
      attachFiltersToSource(voice, blocksData);
    }
  };

  return (
    <div className="modesElement">
      <span className="labels">Sound</span>
      <label htmlFor="toggler" className="switch">
        <input id="toggler" type="checkbox" onClick={onToggler} />
        <span className="slider" />
      </label>
      <span className="labels">Voice</span>
    </div>
  );
};

FilterToggler.propTypes = {
  audioData: PropTypes.instanceOf(Object).isRequired,
  blocksData: PropTypes.instanceOf(Array).isRequired,
  voice: PropTypes.instanceOf(Object),
  sound: PropTypes.instanceOf(Object),
};

const mapStateToProps = state => ({
  audioData: state.audioData,
  blocksData: state.blocksData,
});

export default connect(mapStateToProps)(FilterToggler);
