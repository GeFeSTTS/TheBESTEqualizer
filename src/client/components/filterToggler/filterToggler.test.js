import React from 'react';
import { shallow } from 'enzyme';
import { FilterToggler } from './filterToggler';

describe('Filter Toggler', () => {
  const props = {
    audioData: {
      sound: {
        effects: ['Delay', 'Flanger', 'Reverb', 'Tremolo', 'Stereo panner'],
        removeEffect: jest.fn(),
        addEffect: jest.fn(),
      },
      voice: {
        effects: [],
        removeEffect: jest.fn(),
        addEffect: jest.fn(),
      },
    },
    blocksData: [
      {
        name: 'Delay',
        isVisible: true,
      },
      {
        name: 'Ping-Pong Delay',
        isVisible: false,
      },
      {
        name: 'Dub Delay',
        isVisible: false,
      },
      {
        name: 'Distortion',
        isVisible: false,
      },
      {
        name: 'Quadrafuzz',
        isVisible: false,
      },
      {
        name: 'Flanger',
        isVisible: true,
      },
      {
        name: 'Reverb',
        isVisible: true,
      },
      {
        name: 'Tremolo',
        isVisible: true,
      },
      {
        name: 'Stereo panner',
        isVisible: true,
      },
      {
        name: 'Compressor',
        isVisible: false,
      },
      {
        name: 'Low-Pass Filter',
        isVisible: false,
      },
      {
        name: 'High-Pass Filter',
        isVisible: false,
      },
      {
        name: 'Ring Modulator',
        isVisible: false,
      },
    ],
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<FilterToggler {...props} />);
  });

  it('should render filterToggler element', () => {
    expect(wrapper.find('.modesElement')).toHaveLength(1);
  });

  it('should render filterToggler properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should invoke onToggler function that attach voice filters', () => {
    wrapper.find('#toggler').simulate('click');
    expect(props.audioData.voice.addEffect).toHaveBeenCalled();
  });

  it('should invoke onToggler function that attach sound filters', () => {
    wrapper.find('#toggler').simulate('click');
    wrapper.find('#toggler').simulate('click');
    expect(props.audioData.sound.addEffect).toHaveBeenCalled();
  });
});
