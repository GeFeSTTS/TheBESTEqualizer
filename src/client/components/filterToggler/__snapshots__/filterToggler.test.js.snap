// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Toggler should render filterToggler properly 1`] = `ShallowWrapper {}`;
